elephant/joint_isi_dithering_class.py

```diff
@@ -0,0 +1,438 @@
+        The unit of the spiketrain in the output.
+        Default: pq.s
+    window_length: pq.Quantity
+        The extent in which the joint-ISI-distribution is calculated.
+        Default: 120*pq.ms
+    num_bins: int
+        The size of the joint-ISI-distribution will be num_bins*num_bins.
+        Default: 120
+    sigma: pq.Quantity
+        The standard deviation of the Gaussian kernel, with which
+        the data is convoluted.
+        Default: 0.001*pq.s
+    isi_median_threshold: pq.Quantity
+        Only if the median of the ISI distribution is smaller than
+        isi_median_threshold the Joint-ISI dithering is applied, if not the
+        uniform dithering is used.
+        Default: 30*pq.ms
+    alternate: boolean
+        If alternate == True: then first all even and then all odd spikes are
+        dithered. Else: in ascending order from the first to the last spike, all
+        spikes are moved.
+        Default: True.
+    print_mode: boolean
+        If True, also the way of how the dithered spikes are evaluated
+        is returned so 'uniform' for uniform and dithering and 'jisid' for
+        joint-ISI-dithering
+        Default: False
+    use_sqrt: boolean
+        if use_sqrt == True a sqrt is applied to the joint-ISI histogram,
+        following Gerstein et al. 2004
+        Default: False
+    method: string
+        if 'fast' the entire diagonals of the joint-ISI histograms are
+        used if 'window' only the values of the diagonals are used, whose
+        distance is lower than dither
+        Default: 'fast'
+    cutoff: boolean
+        if True than the Filtering of the Joint-ISI histogram is
+        limited to the lower side by the minimal ISI.
+        This can be necessary, if in the data there is a certain dead time,
+        which would be destroyed by the convolution with the 2d-Gaussian
+        function.
+        Default: True
+    min_spikes: int
+        if the number of spikes is lower than this number, the spiketrain
+        is directly passed to the uniform dithering.
+        Default: 10
+
+    Methods
+    ----------
+    preprocessing()
+        The preprocessing function is called in the initialization process.
+        Outside of it is only necessary if the attributes of the
+        :class:`Joint_ISI_Space` were changed after the initialization, than it
+        prepares the class again to create dithered spiketrains.
+    dithering()
+        Returns a list of dithered spiketrains and if print_mode it returns
+        also a string 'uniform' or 'jisid' indicating the way, how the dithered
+        spiketrains were obtained.
+
+    Output:
+    dithered_sts
+        List of spiketrains, dithered with Joint-ISI-dithering
+    mode (only if print_mode=True)
+        string: Indicates, which method was used to dither the spikes.
+            'jisih' if joint-ISI was used,
+            'uniform' if the dense_rate was too low and uniform dithering was
+             used.
```

Comment:
I guess this is the output of the dithering method. Then I am not sure it needs to go here, it's already in the doc string of the function.